airflow-core/src/airflow/configuration.py

```diff
@@ -873,12 +874,14 @@ def mask_secrets(self):
                 continue
             mask_secret(value)
 
-    def _env_var_name(self, section: str, key: str) -> str:
-        return f"{ENV_VAR_PREFIX}{section.replace('.', '_').upper()}__{key.upper()}"
+    def _env_var_name(self, section: str, key: str, team_id: str | None = None) -> str:
+        team_component = f"{team_id.upper()}__" if team_id else ""
+        return f"{ENV_VAR_PREFIX}{team_component}{section.replace('.', '_').upper()}__{key.upper()}"
```

Comment:
Suggestion: maybe a better approach for env var will be to make sure that the team settings are having a separate namespace:

``AIRFLOW_TEAM__TEAM_ID__SECTION__CONFIG``

There is a non-zero chance someone names their team "core" for example.